app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaFotoActivity.java

```diff
@@ -273,11 +269,13 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                     //recupera todos os dados não alterados e os reescreve no FirebaseDatabase e também recupera o caminho da foto alterada e a atualiza no FirebaseDatabase
                     minhaReceita.adicionarUrlFotoFirebaseDb(idReceitaRecuperada);
 
-                    //Retorna para a main activity após o usuário selecionar uma foto
-                    Intent i = new Intent(NovaReceitaFotoActivity.this, MainActivity.class);
-                    startActivity(i);
-
                     mensagemFotoAdicionada();
+
+                    //encerra a activity anterior ao salvar a foto da receita
+                    NovaReceitaInfoActivity.addInfoReceita.finish();
+
+                    finish();
```

Comment:
Código repetido, isolar em uma função única

Lembrar sempre do princípio DRY (Don't Repeat Yourse): https://dzone.com/articles/software-design-principles-dry-and-kiss